### PR TITLE
JavaScript: WeatherApp

### PR DESCRIPTION
**Pull request by gurbuzertunga**
In this project, I built a Weather app with javascript and webpack. with the use of openweather API.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @gurbuzertunga  👋

**You've done pretty well here.**

- [x] Please refer to comments on your code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "eslint-plugin-import": "^2.20.2",
+    "style-loader": "^2.0.0",
+    "webpack": "^5.11.1",
+    "webpack-cli": "^4.3.0"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "watch": "webpack --watch",
+    "build": "webpack"
+  },
```
Comment (mikenath223) [inline package.json]: - [x] Kindly add other scripts here:
```
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },

```

File: src/utils/converter.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      el.defUnit = 'fahrenheit';
+      el.degreeType.textContent = 'Fahrenheit';
+    } else {
+      el.defValue = parseInt((el.defValue - 32.0) * 5.0 / 9.0);
+      el.degree.textContent = `${el.defValue}°`;
+      el.defUnit = 'celcius';
+      el.degreeType.textContent = 'Celcius';
+    }
+  });
+}
```
Comment (mikenath223) [inline src/utils/converter.js]: - [x] Change this to es6 arrow function syntax.

File: src/utils/converter.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable  */
```
Comment (mikenath223) [inline src/utils/converter.js]: - [x] Please avoid disabling eslint rule for an entire module. Kindly remove this line.

**Comment by mikenath223** [comment]
- [x] OPTIONAL - There were a lot of other info gotten from the API. Consider displaying a few more of them.

**Comment by mikenath223** [comment]
- [x] OPTIONAL - You can consider getting the geolocation of the user and displaying that as the default weather.

**Comment by mikenath223** [comment]
- [x] OPTIONAL - While retrieving data from the API showing a loading animation would be nice.

**Comment by mikenath223** [comment]
- [x] IMPORTANT - Please change the background image on different weather conditions.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi again @gurbuzertunga 👋

**Good job fixing some issues raised. 👍**

- [x] However running the project doesn't work now. After entering a city and clicking the search icon it returns no action and opening the console shows a bunch of errors. Kindly check and fix.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: src/utils/populate-dom.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+      el.weatherOutput.appendChild(el.cityName);
+      el.weatherOutput.appendChild(el.degree);
+      el.weatherOutput.appendChild(el.iconContainer);
+      el.iconContainer.appendChild(el.iconImg);
+
+      el.input.value = '';
+      el.defUnit = 'celcius';
+    });
+  });
```
Comment (mikenath223) [inline src/utils/populate-dom.js]: - [x] Sorry this wasn't mentioned in the previous review, but please chain a `.catch` to your fetch request for error handling in the async request,

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi @gurbuzertunga  👋👋

- [x] It still throws the same error. To give you a better context here is the error: `populate-dom.js:14 Uncaught (in promise) TypeError: Cannot set property defValue of #<Object> which has only a getter
    at eval (populate-dom.js:14)`. It seems to be from populate-dom.js line 14. Kindly check.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Comment by gurbuzertunga** [comment]
> ### Hi @gurbuzertunga 
> * [x]  It still throws the same error. To give you a better context here is the error: `populate-dom.js:14 Uncaught (in promise) TypeError: Cannot set property defValue of #<Object> which has only a getter at eval (populate-dom.js:14)`. It seems to be from populate-dom.js line 14. Kindly check.
> 
> ### **STATUS: CHANGES REQUESTED** 
> Kindly fix these issues and request another review
> Feel free to contact me on slack(michgolden) if you need further clarification.
> 
> HAPPY CODING

Hey! I can't produce the error you are getting on my end. Both local and live demo versions are working error free. I advise you to check on your environment to see the problem.

Attached the working screenshot from it.

Screenshot from 2020-12-30 21-18-04

**Comment by gurbuzertunga** [comment]
I've been looking into the error you've been getting and the error is caused by the object we fetch from Open Weather Api. Since we can't actually modify the object we are fetching from an external server, I don't think this code is supposed to run on dev mode.

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @gurbuzertunga :wave: 

### I can't reproduce the error on my end either, both locally and on your online app. I believe everything is working normally, so your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**